src/main/java/emu/grasscutter/game/chat/ChatSystem.java

```diff
@@ -67,27 +67,27 @@ public void clearHistoryOnLogout(Player player) {
         }
     }
 
-    public void handlePullPrivateChatReq(Player player, int targetUid) {
-        if (this.history.containsKey(player.getUid()) && this.history.get(player.getUid()).containsKey(targetUid)) {
-            player.sendPacket(new PacketPullPrivateChatRsp(this.history.get(player.getUid()).get(targetUid)));
+    public void handlePullPrivateChatReq(Player player, int partnerId) {
+        if (this.history.getOrDefault(player.getUid(), Map.of()).containsKey(partnerId)) {
+            player.sendPacket(new PacketPullPrivateChatRsp(this.history.get(player.getUid()).get(partnerId)));
         }
         else {
             player.sendPacket(new PacketPullPrivateChatRsp(List.of()));
         }
```

Comment:
```suggestion
        var chatHistory = this.history.computeIfAbsent(player.getUid(), HashMap::new)
                                .computeIfAbsent(partnerId, ArrayList::new);
        player.sendPacket(new PacketPullPrivateChatRsp(chatHistory));
```

Reply:
Not sure if applying this directly from web interface is dangerous since it bypassed a UI restriction
![image](https://user-images.githubusercontent.com/5397662/184901766-68ceb97f-baf9-40e2-85f5-4c6669d21100.png)